src/views/Detail/Detail.test.jsx

```diff
@@ -0,0 +1,79 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+
+import { MemoryRouter, Route } from 'react-router-dom';
+
+import Detail from './Detail';
+import Home from '../Home/Home';
+import { charDetail } from '../../utils/test-data';
+
+const server = setupServer(
+  rest.get(
+    'https://bobsburgers-api.herokuapp.com/characters/493',
+    (req, res, ctx) => {
+      return res(ctx.json(charDetail));
+    }
+  )
+);
+
+beforeAll(() => server.listen());
+
+afterAll(() => server.close());
+
+test('checking for image, name, detail, and button to be on the page', async () => {
+  render(
+    <MemoryRouter initialEntries={['/characters/493']}>
+      <Route path="/:select/:id">
+        <Detail />
+      </Route>
+    </MemoryRouter>
+  );
+
+  const name = await screen.findByRole('heading', {
+    name: /warren fitzgerald/i,
+  });
+  expect(name).toBeInTheDocument();
+
+  const image = await screen.findByRole('img');
+  expect(image).toBeInTheDocument();
+
+  const occupation = await screen.findByText(/businessman/i);
+  expect(occupation).toBeInTheDocument();
+
+  const hair = await screen.findByText(/brown/i);
+  expect(hair).toBeInTheDocument();
+
+  const episode = await screen.findByText(/pro tiki/i);
+  expect(episode).toBeInTheDocument();
+
+  const button = await screen.findByRole('button', { name: /back/i });
+  expect(button).toBeInTheDocument();
+});
+
+test('testing to make sure the button takes you back to the main page', async () => {
+  render(
+    <MemoryRouter initialEntries={['/characters/493']}>
+      <Route path="/:select/:id">
+        <Detail />
+      </Route>
+    </MemoryRouter>
+  );
+
+  const button = await screen.findByRole('button', { name: /back/i });
+  expect(button).toBeInTheDocument();
+
+  userEvent.click(button);
+  render(
```

Comment:
I don't know that I would test in this fashion. It might make more sense to render a common parent component for this kind of behavioral testing